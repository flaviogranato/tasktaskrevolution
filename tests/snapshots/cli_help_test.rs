use std::process::Command;

#[test]
fn test_main_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    // Filter out compilation warnings and focus on the help output
    let help_output = if stdout.contains("Usage: ttr") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_create_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "create", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Create new entities") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_create_project_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "create", "project", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Create a new project") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_list_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "list", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("List entities") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}

#[test]
fn test_validate_help() {
    let output = Command::new("cargo")
        .args(&["run", "--", "validate", "--help"])
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);

    let help_output = if stdout.contains("Validate system") {
        stdout
    } else {
        stderr
    };

    insta::assert_snapshot!(help_output);
}
