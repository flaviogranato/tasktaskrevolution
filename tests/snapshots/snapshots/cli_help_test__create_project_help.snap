---
source: tests/snapshots/cli_help_test.rs
expression: help_output
---
Create a new project

Usage: ttr create project [OPTIONS] --name <NAME> --start-date <START_DATE> --end-date <END_DATE>

Options:
  -n, --name <NAME>                    Project name
  -v, --verbose                        Enable verbose output (alias for --debug)
      --code <CODE>                    Project code (optional, will be auto-generated if not provided)
      --debug                          Enable debug output (alias for --verbose)
      --company <COMPANY>              Company code (optional if in company context)
  -q, --quiet                          Enable quiet output (minimal information)
  -d, --description <DESCRIPTION>      Project description
      --start-date <START_DATE>        Start date (YYYY-MM-DD)
      --end-date <END_DATE>            End date (YYYY-MM-DD)
      --template <TEMPLATE>            Template name (optional)
      --template-vars <TEMPLATE_VARS>  Template variables (comma-separated key=value pairs)
  -h, --help                           Print help
